common/djangoapps/student/management/commands/force_send_notification_digest.py

```diff
@@ -0,0 +1,155 @@
+# this must be up at the top otherwise the
+# configuration does not appear to take affect
+import datetime
+import pytz
+from django.conf import settings
+from edx_notifications import const
+from edx_notifications.digests import send_notifications_digest, send_notifications_namespace_digest
+
+LOGGING = {
+    'version': 1,
+    'handlers': {
+        'console': {
+            'class': 'logging.StreamHandler',
+            'stream': sys.stdout,
+        }
+    },
+    'root': {
+        'handlers': ['console'],
+        'level': 'INFO'
+    }
+}
+logging.config.dictConfig(LOGGING)
+
+from django.core.management.base import BaseCommand, CommandError
+
+log = logging.getLogger(__file__)
+
+from optparse import make_option, OptionParser
+
+
+class Command(BaseCommand):
+    """
+    Django management command to force-send the daily/weekly digest emails
+    """
+
+    help = 'Command to force-send the daily/weekly digest emails'
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--daily',
+            action='store_true',
+            dest='send_daily_digest',
+            default=False,
+            help='Force send daily digest email.'
+        ),
+    )
+
+    option_list = option_list + (
+        make_option(
+            '--weekly',
+            action='store_true',
+            dest='send_weekly_digest',
+            default=False,
+            help='Force send weekly digest email.'
+        ),
+    )
+
+    option_list = option_list + (
+        make_option(
+            '--ns',
+            dest='namespace',
+            default='All',
+            help='Specify the namespace. Default = All.'
+        ),
+    )
+
+    def send_daily_digest(self, namespace='All'):
+        """
+        Sends the daily digest.
+        """
+        if const.NOTIFICATION_DIGEST_SEND_TIMEFILTERED:
+            from_timestamp = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=1)
+        else:
+            from_timestamp = None
+
+        to_timestamp = datetime.datetime.now(pytz.UTC)
+        preference_name = const.NOTIFICATION_DAILY_DIGEST_PREFERENCE_NAME
+        subject = const.NOTIFICATION_DAILY_DIGEST_SUBJECT
+        from_email = const.NOTIFICATION_EMAIL_FROM_ADDRESS
+
+        if namespace == "All":
+            digests_sent = send_notifications_digest(from_timestamp, to_timestamp, preference_name, subject, from_email)
+        else:
+            digests_sent = send_notifications_namespace_digest(
+                namespace, from_timestamp, to_timestamp, preference_name, subject, from_email
+            )
+        return digests_sent
+
+    def send_weekly_digest(self, namespace='All'):
+        """
+        Sends the weekly digest.
+        """
+
+        if const.NOTIFICATION_DIGEST_SEND_TIMEFILTERED:
+            from_timestamp = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=7)
+        else:
+            from_timestamp = None
+
+        to_timestamp = datetime.datetime.now(pytz.UTC)
```

Comment:
Several of these values are duplicates of the stuff in daily, as well -- maybe some sort of consolidation?